### ISSUE TITLE
Event handlers for DialogMessageReceived events stop working after an email item is moved.

### ISSUE THREAD

**Issue by andrewlamansky** [OfficeDev/office-js#3050]
> Message from office-js bot:
We’re closing this issue because it has been inactive for a long time. We’re doing this to keep the issues list manageable and useful for everyone. If this issue is still relevant for you, please create a new issue.
> Thank you for your understanding and continued feedback.

Event handlers added in the callback function of `Office.context.ui.displayDialogAsync` will not capture `DialogMessageReceived` events after an email item has been programmatically moved to a different folder in OWA (either by calling the `https://outlook.office365.com/api/v2.0/me/messages/{itemId}/move` API endpoint or by using the `Office.context.mailbox.makeEwsRequestAsync` method).

## Your Environment
* Platform: Web
* Host: Outlook
* Office version number: N/A
* Operating System: Windows 10 Pro
* Browser: Chrome Version 108.0.5359.125 (Official Build) (64-bit)
 
## Expected behavior
When a dialog is opened in an iframe, that dialog should be able to send messages to active event listeners by invoking the `Office.context.ui.messageParent` method.

## Current behavior
When a dialog is opened in an iframe, that dialog _is_ able to send messages to active listeners, _until_ the add-in performs the operation of moving an email to a different folder. This action doesn't impact the dialog's ability to send messages, but it does appear to prevent those messages from being captured by the event handlers wired up in the `Office.context.ui.displayDialogAsync` callback.

## Steps to reproduce
A minimalist Office add-in that demonstrates this behavior is available here:
https://github.com/andrewlamansky/Dialog-API-Bug
The static files in that repository are being served via Github Pages, so once the manifest is installed the add-in is ready to be used. See the repository README for detailed replication instructions.

## Link to live example(s)
https://github.com/andrewlamansky/Dialog-API-Bug

**Comment by exextoc** [comment]
We tried reproing the issue by following the steps mentioned but we are not able to repro. Can you please share a quick video, that will give us better understanding. Thanks.

**Comment by andrewlamansky** [comment]
Please see the attached zip file for a screen capture demonstrating the issue: [dialog-api-bug.zip](https://github.com/OfficeDev/office-js/files/10439692/dialog-api-bug.zip)

Relevant timestamps:
00:14 - add-in button is clicked for the first time
00:18 - add-in's "Move Message to Junk" button is clicked (note info in browser console)
00:33 - add-in button is clicked for the second time
00:38 - add-in's "Move Message to Junk" button is clicked
00:38 - 01:14 - demonstration that the dialog box is sending messages, but those messages are no longer received by the event listeners.
01:10 - demonstration that the second email was not moved to the junk folder.
01:15 - demonstration that first email was successfully moved to junk folder.

**Comment by David-Buyer** [comment]
I have the exact same issue. After moving an email message, the next time i try use Office.context.ui.messageParent() from a dialog on a new message ( activating the add-in from a click button ), it does not send the message to parent anymore. It seems remains hanged to the previous item message context and it does not capture the new one. This does not happen if i do not use the dialog as **displayInIframe: false** neither i do not remove/move the message. I'm struggling to find a solution or a workaround, but with no luck. It would appreciate if someone have a solution. To reproduce the issue is enough to follow the scenario to move an item message after click a dialog popup button, and try again with another item message to do same; the subsequent attempts to Office.context.ui.messageParent() will be broken.

**P.S: this happens on OWA and new Outlook only.**
